### PR TITLE
[WebGPU] Overrides with duplicate @id(values) are not rejected

### PR DESCRIPTION
**Pull request by mwyrzykowski**
#### 8db4245f77a3baa881e5aeae9d80bd900e2af657
<pre>
[WebGPU] Overrides with duplicate @id(values) are not rejected
<a href="https://bugs.webkit.org/show_bug.cgi?id=266259">https://bugs.webkit.org/show_bug.cgi?id=266259</a>
radar://119520604

Reviewed by Dan Glastonbury.

The integer value inside of @id(...) must be unique.

* LayoutTests/http/tests/webgpu/webgpu/api/validation/shader_module/overrides-expected.txt:
* Source/WebGPU/WGSL/AttributeValidator.cpp:
(WGSL::AttributeValidator::visit):
* Source/WebGPU/WGSL/WGSLShaderModule.h:
(WGSL::ShaderModule::containsOverride const):
(WGSL::ShaderModule::addOverride):

Canonical link: <a href="https://commits.webkit.org/271956@main">https://commits.webkit.org/271956@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/e5c7cea48e74f2877ecd0b0f5228c3b574e2e49e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/30034 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/8697 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/31345 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/32533 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27146 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10889 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5948 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27186 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/30328 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/7292 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/25615 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/6229 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/6381 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/26708 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33869 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/27387 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/27139 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32561 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/6323 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/4498 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/30359 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/8071 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7145 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/7075 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6850 "Built successfully") | | |

### CODE REVIEW

File: Source/WebKit/GPUProcess/graphics/RemoteImageBufferSet.cpp [hunk 1]
```diff
@@ -253,6 +253,14 @@ bool RemoteImageBufferSet::makeBuffersVolatile(OptionSet<BufferInSetType> reques
...
 }
 
+#if ENABLE(RE_DYNAMIC_CONTENT_SCALING)
+void RemoteImageBufferSet::dynamicContentScalingDisplayList(CompletionHandler<void(std::optional<WebCore::DynamicContentScalingDisplayList>&&)>&& completionHandler)
+{
+    auto displayList = m_frontBuffer->dynamicContentScalingDisplayList();
+    completionHandler({ WTFMove(displayList) });
+}
+#endif
+
```
Comment (djg) [inline Source/WebKit/GPUProcess/graphics/RemoteImageBufferSet.cpp]: Is this an unrelated change?

**Review by djg — APPROVED** [review]
:shipit:

File: Source/WebKit/GPUProcess/graphics/RemoteImageBufferSet.cpp (hunk 1 above)
Comment (mwyrzykowski) [reply Source/WebKit/GPUProcess/graphics/RemoteImageBufferSet.cpp]: Yes, will revert. `git checkout origin/main $file` but my origin was stale :(
